pages/index.vue

```diff
@@ -688,17 +710,46 @@
 
         this.loading = false
       },
-      updatePlatformFilter(value) {
-        this.paramPlatform = value
-        this.getSuites()
-      },
-      updateVersionFilter(value) {
-        this.paramVersion = value
+      updateFilers() {
+        const query = {
+          ...this.$route.query,
+          campaign: this.paramCampaign,
+          platform: this.paramPlatform,
+          version: this.paramVersion
+        }
+        if (this.paramCampaign === 'All') {
+          delete query.campaign
+        }
+        if (this.paramPlatform === 'All') {
+          delete query.platform
+        }
+        if (this.paramVersion === 'All') {
+          delete query.version
+        }
+        this.$router.push({ query })
         this.getSuites()
       },
-      updateCampaignFilter(value) {
-        this.paramCampaign = value
-        this.getSuites()
+      updatePagination() {
+        const pagination = {
+          ...this.$route.query,
+          descending: this.pagination.descending,
+          page: this.pagination.page,
+          rowsPerPage: this.pagination.rowsPerPage,
+          sortBy: this.pagination.sortBy
+        }
+        if (this.pagination.descending === true) {
+          delete pagination.descending
+        }
+        if (this.pagination.page === 1) {
+          delete pagination.page
+        }
+        if (this.pagination.rowsPerPage === 20) {
+          delete pagination.rowsPerPage
+        }
+        if (this.pagination.sortBy === 'date') {
+          delete pagination.sortBy
+        }
```

Comment:
I would invert this, in case it's not the value you want, add the property

`delete` can be sometime a bit expensive to use